app/modules/collaborations/models.py

```diff
@@ -17,13 +51,126 @@ class Collaboration(db.Model, HoustonModel):
     guid = db.Column(
         db.GUID, default=uuid.uuid4, primary_key=True
     )  # pylint: disable=invalid-name
-    title = db.Column(db.String(length=50), nullable=False)
+    title = db.Column(db.String(length=50), nullable=True)
+
+    collaboration_user_associations = db.relationship(
+        'CollaborationUserAssociations', back_populates='collaboration'
+    )
+
+    def __init__(
+        self, user_guids=None, approval_states=None, is_initiator=None, **kwargs
+    ):
+        # I think this can be adapted to allow >2 User collabs, opening the door for Org-level.
+        if user_guids is None or len(user_guids) != 2:
+            raise ValueError(
+                '__init__ Collaboration: Creating a collaboration requires a user_guids list.'
+            )
+
+        # You can create a collaboration where both users 'approved'
+        if approval_states is not None and len(approval_states) != len(user_guids):
+            raise ValueError(
+                '__init__ Collaboration: Length of approval_states was not equal to provided user list.'
+            )
+
+        if is_initiator is not None and len(is_initiator) != len(user_guids):
+            raise ValueError(
+                '__init__ Collaboration: Length of is_initiator states was not equal to provided user list.'
+            )
+
+        for i in range(len(user_guids)):
+            from app.modules.users.models import User
+
+            user_tup = User.ensure_edm_obj(user_guids[i])
+
+            if user_tup is None or user_tup[0] is None:
+                raise ValueError(
+                    '__init__ Collaboration: One of the user_guids provided was had no associated user.'
+                )
+
+            collab_user_assoc = CollaborationUserAssociations(
+                collaboration=self, user=user_tup[0]
+            )
+
+            if (
+                approval_states is not None
+                and approval_states[i] in CollaborationUserState.ALLOWED_STATES
+            ):
+                collab_user_assoc.read_approval_state = approval_states[i]
+            if is_initiator is not None and isinstance(is_initiator[i], bool):
+                collab_user_assoc.initiator = is_initiator[i]
+                # If you initiate you approve
+                collab_user_assoc.read_approval_state = CollaborationUserState.APPROVED
+
+            with db.session.begin():
+                db.session.add(collab_user_assoc)
+                db.session.merge(user_tup[0])
+
+    def get_users(self):
```

Comment:
Should we be having these get_ type interfaces or the property type interfaces as in organisation?
https://github.com/WildMeOrg/houston/blob/develop/app/modules/organizations/models.py#L152
I think we should be doing the latter

Reply:
i have kind of wondered about this myself.  curious what @bluemellophone might have to say about this?  we have plenty of examples of `get_` ... but plenty of the other.  isnt it the case that property-type you drop the `()` when calling?  i feel like there was some other reasons to use property, that had to do with schema -- like a property can be used in them [like i did with assets](https://github.com/WildMeOrg/houston/blob/develop/app/modules/assets/schemas.py#L40).

so i can see reasons to do both, i guess.

Reply:
Sounds good. I'll change these to match the pattern in organizations.